### ISSUE TITLE
Problem with broken images in wrapping carousels

### ISSUE THREAD

**Issue by maredudd** [bbc/tal#449]
Is there a good way to handle broken images in wrapping carousels? In Chrome I can handle this with the `background-image` and using CSS to hide the broken image icon. However, this doesn't work in Amazon Fire TV.  To get round the problem I added an event listener to the image and if there's an error I can set the image to the default image.

However, in a wrapping carousel `WrappingStrip` creates clones of the carousel elements, but does not clone the event listeners. This results in broken images still visible.

One solution would be to clone event listeners as well, and another would be to find a way to get this to work with CSS in Fire TV, but I haven't found one yet.

**Comment by dhurrell** [comment]
Interesting problem!

We use the event listener approach for placeholder images ourselves. We haven't had the same problem, as we don't have any wrapping carousels that contain images, but it makes sense that the event listeners aren't cloned (looking at the code in `wrappingstrip.js`, it ultimately calls through to the DOM API's [Node.cloneNode()](https://developer.mozilla.org/en/docs/Web/API/Node/cloneNode) - see the note on cloning event listeners).

If the CSS approach seems like a dead end, probably the simplest fix is to make your own derivation of `wrappingstrip.js` in your own app, which overrides [_cloneElement()](https://github.com/bbc/tal/blob/7.4.0/static/script/widgets/carousel/strips/wrappingstrip.js#L259-L265) with an implementation that copies the onError event handler to the cloned element. I think your case is too specific to put a fix in TAL itself, but feel free to disagree!

**Comment by maredudd** [comment]
I'm not convinced that our case is too specific since I imagine that there are nearly always images in wrapping strips and it turns out that we couldn't remove the broken image icon on a couple of other platforms either. 

We managed to remove it on Fire TV with `text-indent: -1000px;` but I think there was still a problem with Samsung Tizen and I can't remember which other device.

Anyway, we got round it easily by overriding `_cloneElement()` in our own wrapping strip so this problem is solved for us, so I'm happy for this issue to be closed.
